src/FirebaseAuth.js

```diff
@@ -23,9 +23,13 @@
 
   FirebaseAuth = function($firebaseUtils, auth) {
     this._utils = $firebaseUtils;
-    if (typeof ref === 'string') {
-      throw new Error('Please provide a Firebase reference instead of a URL when creating a `$firebaseAuth` object.');
+
+    if (typeof auth === 'string') {
+      throw new Error('The $firebaseAuth service accepts a Firebase auth instance (or nothing) instead of a URL.');
+    } else if (typeof auth.ref !== 'undefined') {
+      throw new Error('The $firebaseAuth service accepts a Firebase auth instance (or nothing) instead of a Database reference.');
```

Comment:
Yeah, so we have `firebase.database` and `firebase.database.Database`.  So being specific may help here.